File: ruby/get_my_ip.rb
#!/usr/bin/env ruby
# get_my_ip.rb - Fetch external IP and headers from curlmyip.ru

require 'net/http'
require 'json'

BASE_URL = 'http://curlmyip.ru'

def fetch_from_curlmyip(endpoint)
  uri = URI("#{BASE_URL}#{endpoint}")
  response = Net::HTTP.get_response(uri)
  if response.is_a?(Net::HTTPSuccess)
    response.body.strip
  else
    puts "HTTP error: #{response.code} #{response.message}"
    nil
  end
rescue StandardError => e
  puts "Network error: #{e.message}"
  nil
end

# Fetch IP address
ip = fetch_from_curlmyip('/ip')
puts "Your IP address: #{ip}" if ip

# Fetch User-Agent
ua = fetch_from_curlmyip('/ua')
puts "Your User-Agent: #{ua}" if ua

# Fetch and display headers
headers_json = fetch_from_curlmyip('/headers')
if headers_json
  headers = JSON.parse(headers_json)
  puts "Headers sent by you:"
  headers.each { |key, value| puts "  #{key}: #{value}" }
end

File: php/get_my_ip.php
#!/usr/bin/env php
<?php
// get_my_ip.php - Fetch external IP and other information from curlmyip.ru

$base_url = "http://curlmyip.ru";

// Fetch IP address
$ip = file_get_contents("$base_url/ip");
if ($ip !== false) {
    echo "Your IP address: " . trim($ip) . PHP_EOL;
} else {
    echo "Failed to get IP address." . PHP_EOL;
}

// Fetch User-Agent
$ua = file_get_contents("$base_url/ua");
if ($ua !== false) {
    echo "Your User-Agent: " . trim($ua) . PHP_EOL;
}

// Fetch and display headers (JSON)
$headers = file_get_contents("$base_url/headers");
if ($headers !== false) {
    $decoded = json_decode($headers, true);
    if ($decoded) {
        echo "Headers sent by you:" . PHP_EOL;
        foreach ($decoded as $key => $value) {
            echo "  $key: $value" . PHP_EOL;
        }
    } else {
        echo "Raw headers response: $headers" . PHP_EOL;
    }
}
?>